tests/Homework 17/registration_pom.spec.js

```diff
@@ -0,0 +1,293 @@
+import WelcomePage from "../pageObject/WelcomePage.js";
+import SignInPopup from "../components/SignInPopup.js";
+import RegistrationPopup from "../components/RegistrationPopup.js";
+import {expect, test} from "@playwright/test";
+
+test.describe('Registration', async () => {
+    let page
+    let welcomePage
+
+    test.beforeEach(async ({browser}) => {
+        page = await browser.newPage()
+        welcomePage = new WelcomePage(page)
+        await welcomePage.visit()
+
+    })
+
+    test('Registration with correct data', async () => {
+
+        await welcomePage.signInButton.click()
+        const signInPopup = new SignInPopup(page)
+        await signInPopup.registrationButton.click()
+        const registrationPopup = new RegistrationPopup(page)
+
+        await registrationPopup.inputName.fill('Olena')
+        await registrationPopup.inputLastName.fill('Nikitina')
+        await registrationPopup.inputEmail.fill(registrationPopup.generateRandomEmail())
+        await registrationPopup.inputPassword.fill('1321@Helena')
+        await registrationPopup.inputPasswordConfirmation.fill('1321@Helena')
+        await registrationPopup.registerButton.click()
+    })
+
+    test('Name validation - number of symbols', async () => {
+
+        await welcomePage.signInButton.click()
+        const signInPopup = new SignInPopup(page)
+        await signInPopup.registrationButton.click()
+        const registrationPopup = new RegistrationPopup(page)
+
+        await registrationPopup.inputName.fill('a')
+        await registrationPopup.inputName.blur()
+        await expect(registrationPopup.errorMessage).toHaveText('Name has to be from 2 to 20 characters long')
+        await registrationPopup.inputName.clear()
+
+    })
+
+
+    test('Name validation - required field', async () => {
+
+        await welcomePage.signInButton.click()
+        const signInPopup = new SignInPopup(page)
+        await signInPopup.registrationButton.click()
+
+        const registrationPopup = new RegistrationPopup(page)
```

Comment:
it would be nice to return such components from PO so you can decrease imports number